lib/saml/url_service.rb

```diff
@@ -39,7 +39,7 @@ def initialize(saml_settings, session: nil, user: nil, params: {})
 
     # REDIRECT_URLS
     def base_redirect_url
-      VIRTUAL_HOST_MAPPINGS[current_host][:base_redirect]
+      Settings.saml.relay || VIRTUAL_HOST_MAPPINGS[current_host][:base_redirect]
```

Comment:
1) [Here](https://github.com/department-of-veterans-affairs/devops/blob/775cc72c84e13f5b93b8fcb3fc862bc514a3c30c/ansible/roles/review-instance-configure/vars/settings.local.yml#L103) is the `Settings.saml.relay` value (right?).
This being the `base_redirect_url`, shouldn't it be the base URL and not contain the `/auth/login/callback` path since that is provided by the `LOGIN_REDIRECT_PARTIAL` constant? 

2) Can this data be configured to be part of the `VIRTUAL_HOST_MAPPINGS` constant instead of using this `||` operator?  I know the chances are probably very low, but that would prevent this from breaking in the case where `Settings.saml.relay` is erroneously present in `prod`, `staging`, or `dev`

Reply:
1. Yes. And yes! Definitely an oversight by me that I'll correct.
2. This is a good point about `Settings.saml.relay` being erroneously present - unfortunately I don't think it's possible since the base url will be unique for each review instance (e.g. `http://617bed45ccb1fc2a87872b567c721009-api.review.vetsgov-internal`), making it impossible to hardcode into a ruby file.

As a tangent, I don't think _any_ of the dev/stag/prod mappings should be hardcoded into a ruby file. I'd rather see each of these as config in their respective settings-env.yml files, which would eliminate the need for this `def base_redirect_url` method entirely.

Reply:
Good point.  That whole constant is environment specific config.